yt_dlp/extractor/yandexvideo.py

```diff
@@ -254,11 +255,17 @@ class ZenYandexIE(InfoExtractor):
 
     def _real_extract(self, url):
         video_id = self._match_id(url)
-        webpage = self._download_webpage(url, video_id)
-        redirect = self._search_json(r'var it\s*=', webpage, 'redirect', id, default={}).get('retpath')
-        if redirect:
-            video_id = self._match_id(redirect)
-            webpage = self._download_webpage(redirect, video_id, note='Redirecting')
+        redirect_json = self._search_json(r'var it\s*=', webpage, 'redirect', video_id, default={})
+        retpath = redirect_json.get('retpath')
```

Comment:
oops, I removed too much in my last suggestions
```suggestion
        webpage = self._download_webpage(url, video_id)
        redirect_json = self._search_json(r'var it\s*=', webpage, 'redirect', video_id, default={})
        retpath = redirect_json.get('retpath')
```